mujoco_py/mjsim.pyx

```diff
@@ -165,6 +192,31 @@ cdef class MjSim(object):
         self.udd_state = None
         self.step_udd()
 
+    def set_substep_udd_fn(self, substep_udd_fn):
+        '''
+        TODO: tons of docs right here
+        '''
+        if substep_udd_fn is None:
+            self._substep_udd_fn = NULL
+        elif isinstance(substep_udd_fn, int):
```

Comment:
should we ensure that "substep_udd_fn" can be only string, or NULL.

Reply:
It's a bit safer.

Reply:
The reason to allow ints is this lets us re-use these functions between sims (in a single python process).

```
sim1 = MjSim(...., substep_udd_fn=...)
sim2 = MjSim(...., substep_udd_fn=sim1.substep_udd_fn)
```

Reply:
Added test which shows this